File: backend-ms1/src/main/java/ma/zs/stocky/service/facade/admin/crm/ClientAdminService.java
package ma.zs.stocky.service.facade.admin.crm;

import java.util.List;
import ma.zs.stocky.bean.core.crm.Client;
import ma.zs.stocky.dao.criteria.core.crm.ClientCriteria;
import ma.zs.stocky.zynerator.service.IService;



public interface ClientAdminService extends  IService<Client,ClientCriteria>  {

    List<Client> findByClientCategoryId(Long id);
    int deleteByClientCategoryId(Long id);
    long countByClientCategoryCode(String code);




}

File: frontend/src/app/shared/service/admin/crm/ClientAdmin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BehaviorSubject, Observable} from 'rxjs';

import {environment} from 'src/environments/environment';

import {ClientDto} from 'src/app/shared/model/crm/Client.model';
import {ClientCriteria} from 'src/app/shared/criteria/crm/ClientCriteria.model';
import {AbstractService} from 'src/app/zynerator/service/AbstractService';


@Injectable({
  providedIn: 'root'
})
export class ClientAdminService extends AbstractService<ClientDto, ClientCriteria> {
     constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    public constrcutDto(): ClientDto {
        return new ClientDto();
    }

    public constrcutCriteria(): ClientCriteria {
        return new ClientCriteria();
    }

    get API() {
        return environment.apiUrlCollaboratorservice + 'admin/client/';
    }
}
